Bottom: 200a9d6f3734840cccc9509cdd05797325c3f2a9
Top:    f6fa855820df39d0b5714cb1ec7b24be99c85322
Author: Lawrence D'Anna <lawrence_danna@apple.com>
Date:   2019-09-19 13:02:10 -0700

convert SBDebugger::***FileHandle() wrappers to native files.


---
diff --git a/lldb/scripts/interface/SBDebugger.i b/lldb/scripts/interface/SBDebugger.i
index 3851c428bfb..ab4f998379d 100644
--- a/lldb/scripts/interface/SBDebugger.i
+++ b/lldb/scripts/interface/SBDebugger.i
@@ -165,29 +165,41 @@ public:
     void
     SkipLLDBInitFiles (bool b);
 
-    %feature("autodoc", "DEPRECATED, use SetInputFile");
-    void
-    SetInputFileHandle (FILE *f, bool transfer_ownership);
-
-    %feature("autodoc", "DEPRECATED, use SetOutputFile");
-    void
-    SetOutputFileHandle (FILE *f, bool transfer_ownership);
-
-    %feature("autodoc", "DEPRECATED, use SetErrorFile");
-    void
-    SetErrorFileHandle (FILE *f, bool transfer_ownership);
-
-    %feature("autodoc", "DEPRECATED, use GetInputFile");
-    FILE *
-    GetInputFileHandle ();
-
-    %feature("autodoc", "DEPRECATED, use GetOutputFile");
-    FILE *
-    GetOutputFileHandle ();
-
-    %feature("autodoc", "DEPRECATED, use GetErrorFile");
-    FILE *
-    GetErrorFileHandle ();
+    %extend {
+
+        %feature("autodoc", "DEPRECATED, use SetInputFile");
+        void
+        SetInputFileHandle (lldb_private::File &file,
+                            bool transfer_ownership) {
+            self->SetInputFile(file);
+        }
+
+        %feature("autodoc", "DEPRECATED, use SetOutputFile");
+        void
+        SetOutputFileHandle (lldb_private::File &file,
+                             bool transfer_ownership) {
+            self->SetOutputFile(file);
+        }
+
+        %feature("autodoc", "DEPRECATED, use SetErrorFile");
+        void
+        SetErrorFileHandle (lldb_private::File &file,
+                             bool transfer_ownership) {
+            self->SetErrorFile(file);
+        }
+
+        lldb_private::File GetInputFileHandle() {
+            return self->GetInputFile().GetFile();
+        }
+
+        lldb_private::File GetOutputFileHandle() {
+            return self->GetOutputFile().GetFile();
+        }
+
+        lldb_private::File GetErrorFileHandle() {
+            return self->GetErrorFile().GetFile();
+        }
+    }
 
     SBError
     SetInputFile (SBFile file);
